lib/Image/Adapter/Imagick.php

```diff
@@ -58,27 +58,24 @@ public function load($imagePath, $options = [])
         }
 
         // support image URLs
-        if (preg_match('@^https?://|^s3://@', $imagePath)) {
-            $tmpFilename = 'imagick_auto_download_' . md5($imagePath) . '.' . File::getFileExtension($imagePath);
-            $tmpFilePath = PIMCORE_SYSTEM_TEMP_DIRECTORY . '/' . $tmpFilename;
-
-            $this->tmpFiles[] = $tmpFilePath;
-
-            if (preg_match('@^s3://@', $imagePath)) {
-                $imageContent = file_get_contents($imagePath);
+        if (!stream_is_local($imagePath)) {
+            if (isset($options['asset'])) {
```

Comment:
But for locally stored images we actually do not need a temporary file. Or have I understood you wrong?

Reply:
I think you got me wrong :) 
What I meant is that this code does actually the same thing: 
```php
if (!stream_is_local($imagePath)) {
        $tmpFilename = 'imagick_auto_download_'.md5($imagePath).'.'.File::getFileExtension($imagePath);
        $tmpFilePath = PIMCORE_SYSTEM_TEMP_DIRECTORY.'/'.$tmpFilename;

        $this->tmpFiles[] = $tmpFilePath;

        $src = fopen($imagePath, 'rb');
        $dest = fopen($tmpFilePath, 'wb', false, File::getContext());
        stream_copy_to_stream($src, $dest);
        fclose($dest);

        $imagePath = $tmpFilePath;
    }
}
``` 

we don't need this part: 
```php
    if (isset($options['asset'])) {
        // imagick is only able to deal with local files
        // if your're using custom stream wrappers this wouldn't work, so we create a temp. local copy
        $imagePath = $options['asset']->getTemporaryFile();
    } else {
``` 

because if `$imagePath` isn't a local stream we have to create a temp file anyway 😊

Reply:
Ok, I removed this condition.

Reply:
Ok, implemented for all files you have mentioned except https://github.com/pimcore/pimcore/blob/f093a1893b83495bc864ea5d89d25a42b15e8a72/lib/Image/Optimizer.php#L101
There I did not understand where a local file is necessary.

Reply:
You're right, https://github.com/pimcore/pimcore/blob/f093a1893b83495bc864ea5d89d25a42b15e8a72/lib/Image/Optimizer.php#L101 is fine as it is.